io-engine/src/bdev/nexus/nexus_io.rs

```diff
@@ -642,7 +642,8 @@ impl<'n> NexusBio<'n> {
             // set the IO as failed in the submission stage.
             self.ctx_mut().failed += 1;
 
-            self.channel_mut().disconnect_device(&device);
+            self.channel_mut().detach_device(&device);
+            self.channel_mut().disconnect_detached_devices(|_| true);
```

Comment:
hmm interesting, why do we disconnect here when we fail to submit?

Reply:
But isn't this bypassing the whole fault_child logic? We're removing the device before we persist to etcd?

Reply:
It was like this before. I created a ticket to investigate and address this potential problem.